File: 0070-climbing-stairs/0070-climbing-stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        int prev2=0;
        int prev1=1;
        for(int i=1;i<=n;i++){
            int curr=prev2+prev1;
            prev2=prev1;
            prev1=curr;
        }
        return prev1;
    }
};

File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n<= 3:
            return n
        
        prev1 , prev2 = 3,2
        for _ in range(3,n):
            curr = prev1 + prev2
            prev2=prev1
            prev1=curr
        return curr